web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx

```diff
@@ -1,32 +1,169 @@
+import { useEffect, useState } from 'react';
+
 import {
+  Button,
+  Input,
   PopupModal,
   PopupModalContent,
   PopupModalHeader,
 } from '@massalabs/react-ui-kit';
+import { AxiosError } from 'axios';
+import { FiPlus } from 'react-icons/fi';
+import { useParams } from 'react-router-dom';
 
-import { ImportForm } from '.';
+import { InputsErrors, assetImportErrors } from '@/const/assets/assets';
+import { usePost, useResource } from '@/custom/api';
 import Intl from '@/i18n/i18n';
+import { IToken } from '@/models/AssetModel';
+import { ImportResult } from '@/pages/Assets';
 
 export function AssetsImportModal({ ...props }) {
   const { closeModal } = props;
 
+  const [importResult, setImportResult] = useState<boolean | null>(null);
+  const [inputError, setInputError] = useState<InputsErrors | null>(null);
+  const [tokenAddress, setTokenAddress] = useState<string>('');
+
+  const { nickname } = useParams();
+
+  const { refetch } = useResource<IToken[]>(`accounts/${nickname}/assets`);
+
+  const {
+    mutate,
+    data,
+    isSuccess: postSuccess,
+    isError: postError,
+    error,
+  } = usePost<IToken>(
+    `accounts/${nickname}/assets?assetAddress=${tokenAddress}`,
+  );
+  const axiosError = error as AxiosError;
+  const postErrorStatus = axiosError?.response?.status;
+
+  useEffect(() => {
+    if (postSuccess) {
+      refetch();
+      setImportResult(true);
+    } else if (postError) {
+      setImportResult(false);
+      displayErrors(postErrorStatus);
+    }
+  }, [postSuccess, postError]);
+
+  function displayErrors(postStatus: number | undefined) {
+    switch (postStatus) {
+      case assetImportErrors.badRequest:
+        setInputError({
+          address: Intl.t('assets.import.failure-screen.bad-request'),
+        });
+        break;
+      case assetImportErrors.invalidAddress:
+        setInputError({
+          address: Intl.t('assets.import.failure-screen.invalid-address'),
+        });
+        break;
+      case assetImportErrors.notFound:
+        setInputError({
+          address: Intl.t('assets.import.failure-screen.not-found'),
+        });
+        break;
+      case assetImportErrors.serverError:
+        setInputError({
+          address: Intl.t('assets.import.failure-screen.internal-server-error'),
+        });
+        break;
+      default:
+        setInputError({
+          address: Intl.t('assets.import.failure-screen.unkown-error'),
+        });
+    }
+  }
+  function isValidAssetAddress(input: string): boolean {
+    const regexPattern = /^AS[0-9a-zA-Z]+$/;
+    return regexPattern.test(input);
+  }
```

Comment:
this function should not be here. We have a collection of validation functions here https://github.com/massalabs/station-massa-wallet/tree/6c317bd32a8608bf55ce9d772acf4092931b0e83/web-frontend/src/validation

Reply:
moved it to the right spot !